From d3aff6a9f47f2828a0c9c78644821885748f8c78 Mon Sep 17 00:00:00 2001
From: Abhi Teja Cholleti <achollet@qti.qualcomm.com>
Date: Mon, 13 Oct 2025 15:43:50 +0530
Subject: [PATCH 1/1] PWHM: DTIMPeriod add validation range

Added validation check to ensure DTIMPeriod is within 1–255 range.
Updated header comment to reflect valid range and default value

Signed-off-by: Abhi Teja Cholleti <achollet@qti.qualcomm.com>
---
odl/wld_radio.odl | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/odl/wld_radio.odl b/odl/wld_radio.odl
index 638d8d4..1be15fb 100644
--- a/odl/wld_radio.odl
+++ b/odl/wld_radio.odl
@@ -1020,9 +1020,13 @@ define probelist_t{
 		* of Beacon frames containing a TIM element whose DTIM Count field is 0. This parameter
 		* is based on dot11DTIMPeriod from [802.11-2012].
 		*
+		* Valid range: 1 to 255
+		* Default value: 3
+		*
 		* @version 9.2
 		*/
-		%persistent uint32 DTIMPeriod{
+		%persistent uint8 DTIMPeriod{
+			on action validate call check_range { min = 1, max = 255 };
 			default 3;
 		}
 
-- 
2.34.1

